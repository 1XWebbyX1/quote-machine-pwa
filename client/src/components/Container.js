import React from 'react'
import Wrapper from './Wrapper/Wrapper.js'

class Container extends React.Component{
  render() {
    return(
        <div  id="container">
           <Wrapper id="quote-box" />
        </div>
    )
  }
}
export default Container;
