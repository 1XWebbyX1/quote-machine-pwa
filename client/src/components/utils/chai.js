import chai from 'chai';
import chaiEnzyme from 'chai-enzyme';
import chaiJestSnapshot from "chai-jest-snapshot";

chai.use(chaiJestSnapshot);
chai.use(chaiEnzyme());
chaiJestSnapshot.resetSnapshotRegistry();

export const expect = chai.expect;
