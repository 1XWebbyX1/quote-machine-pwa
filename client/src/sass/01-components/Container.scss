#container{
  position: relative;
  margin-top: 10%;
  left: 50%;
  transform: translateX(-50%);
  display: flex;
  justify-content: center !important;
  align-items: center;
  resize: both;
  width: calc(50vw + 2em);
  height: calc(35vw + 2em);
  background: rgba(0, 0, 0, 0.3);

  .card-wrapper{
    display: flex;
    justify-content: center;
    align-items: center;
    resize: both;
    width: 90%;
    height: 90%;
    border: 2px solid white;
    background: transparent;

    .hide{
      opacity: 0;
    }

    .show{
      opacity: 1;
    }

    .card{
      padding: 0 20px;
      width: 90%;
      height: 90%;
      transition: opacity 2s ease;
      background: rgba(0, 0, 0, 0.3);

      #text{
        margin: 10% auto;
        font-size: calc(0.5em + 1.5vw);
        text-align: center;
        color: white;
        line-height: 1;
        margin-bottom: 4%;
        user-select: all;
        cursor: default;
      }

      #Q{
        font-size: calc(0.5em + 3.5vw);
        color: $color;
        user-select: none;
      }

      #subs{
        margin: 0% 2%;
        padding: 8px;
        display: flex;
        justify-content: space-between;
        background: rgba(0, 0, 0, 0.3);

        #author{
        user-select: none;
        text-align: right;
        color: $color;
        font-size: calc(0.5em + 1.3vw);
        line-height: 1.1;
        }

        #icons{
          width: 20%;
          display: flex;
          justify-content: space-between;

          #copy-quote{
          color: $color;
          font-size: calc(0.5em + 1.5vw);
          cursor: pointer;
          }

          #tweet-quote{
          color: $color;
          font-size: calc(0.5em + 1.5vw);
          margin-left: -50%;
          cursor: pointer;
          }
        }


      }

    }
  }
}
