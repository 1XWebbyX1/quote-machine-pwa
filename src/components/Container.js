import React from 'react'
import asyncComponent from './asyncComponent/async'

const Wrapper = asyncComponent(() =>
     import('./Wrapper').then(module => module.default)
);

class Container extends React.Component{
  render() {
    return(
        <div  id="container">
           <Wrapper id="quote-box" />
        </div>
    )
  }
}
export default Container;
