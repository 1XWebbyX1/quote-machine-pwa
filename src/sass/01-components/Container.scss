#container{
  position: relative;
  margin: auto;
  margin-top: 10%;
  display: flex;
  justify-content: center;
  align-items: center;
  resize: both;
  width: calc(50vmax + 2rem);
  height: calc(35vmax + 2rem);
  background: rgba(0, 0, 0, 0.3);

  .card-wrapper{
    margin: auto;
    display: flex;
    justify-content: center;
    align-items: center;
    resize: both;
    width: 90%;
    height: 90%;
    border: 2px solid white;
    background: transparent;

    .card{
      margin: auto;
      padding: 0 20px;
      width: 90%;
      height: 90%;
      background: rgba(0, 0, 0, 0.3);

      #text{
        margin: 10% auto;
        font-size: 2vmax;
        text-align: center;
        color: white;
        line-height: 1.1;
        margin-bottom: 4%;
        user-select: none;
      }

      #Q{
        font-size: 4vmax;
        color: $color;
        user-select: none;
      }

      #subs{
        margin: 0% 2%;
        padding: 8px;
        display: flex;
        justify-content: space-between;
        background: rgba(0, 0, 0, 0.3);

        #author{
        user-select: none;
        text-align: right;
        color: $color;
        font-size: 1.8vmax;
        line-height: 1.1;
        }

        #icons{
          width: 20%;
          display: flex;
          justify-content: space-between;

          #copy-quote{
          color: $color;
          font-size: 2vmax;
          cursor: pointer;
          }

          #tweet-quote{
          color: $color;
          font-size: 2vmax;
          margin-left: -50%;
          cursor: pointer;
          }
        }


      }

    }
  }
}
