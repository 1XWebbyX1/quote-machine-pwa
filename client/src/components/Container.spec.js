import React from 'react';
// Our Dependencies
import { expect } from './utils/chai';
import renderer from "react-test-renderer";

// Our Component
import Container from './Container';

describe('Container', () => {

  it("renders correctly", () => {
     const output = renderer.create(
         <Container />
   ).toJSON();
     expect(output).to.matchSnapshot();
  });
})
