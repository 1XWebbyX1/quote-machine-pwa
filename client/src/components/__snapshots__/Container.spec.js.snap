// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders correctly 1`] = `
<div
  id="container"
>
  <div
    className="card-wrapper"
  >
    <div
      className="card show"
    >
      <div
        id="Q"
      >
        Q
      </div>
      <div
        id="text"
      >
        Too many of us are not living our dreams because we are living our fears.
      </div>
      <span
        className="tooltiptext hide"
      >
        Copied!
      </span>
      <div
        id="subs"
      >
        <div
          id="icons"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-clipboard fa-w-12 "
            data-icon="clipboard"
            data-prefix="fas"
            id="copy-quote"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 384 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 112v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V112c0-26.51 21.49-48 48-48h80c0-35.29 28.71-64 64-64s64 28.71 64 64h80c26.51 0 48 21.49 48 48zM192 40c-13.255 0-24 10.745-24 24s10.745 24 24 24 24-10.745 24-24-10.745-24-24-24m96 114v-20a6 6 0 0 0-6-6H102a6 6 0 0 0-6 6v20a6 6 0 0 0 6 6h180a6 6 0 0 0 6-6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-twitter-square fa-w-14 "
            data-icon="twitter-square"
            data-prefix="fab"
            id="tweet-quote"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-48.9 158.8c.2 2.8.2 5.7.2 8.5 0 86.7-66 186.6-186.6 186.6-37.2 0-71.7-10.8-100.7-29.4 5.3.6 10.4.8 15.8.8 30.7 0 58.9-10.4 81.4-28-28.8-.6-53-19.5-61.3-45.5 10.1 1.5 19.2 1.5 29.6-1.2-30-6.1-52.5-32.5-52.5-64.4v-.8c8.7 4.9 18.9 7.9 29.6 8.3a65.447 65.447 0 0 1-29.2-54.6c0-12.2 3.2-23.4 8.9-33.1 32.3 39.8 80.8 65.8 135.2 68.6-9.3-44.5 24-80.6 64-80.6 18.9 0 35.9 7.9 47.9 20.7 14.8-2.8 29-8.3 41.6-15.8-4.9 15.2-15.2 28-28.8 36.1 13.2-1.4 26-5.1 37.8-10.2-8.9 13.1-20.1 24.7-32.9 34z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          id="author"
        >
          Les Brown
        </div>
      </div>
      <button
        disabled={false}
        id="new-quote"
        onClick={[Function]}
      >
        New  Quote
      </button>
    </div>
  </div>
</div>
`;
